src/phatjs.rs

```diff
@@ -0,0 +1,142 @@
+#![cfg_attr(not(feature = "native"), no_main)]
+
+extern crate alloc;
+
+use js::ToJsValue;
+
+use anyhow::{anyhow, bail, Context, Result};
+use sidevm_quickjs::{runtime, Service};
+
+use pink_types::js::{JsCode, JsValue};
+
+struct Args {
+    codes: Vec<JsCode>,
+    js_args: Vec<String>,
+}
+
+fn parse_args() -> Result<Args> {
+    let mut codes = vec![];
+    let mut iter = std::env::args();
+    iter.next();
+    while let Some(arg) = iter.next() {
+        if arg.starts_with("-") {
+            if arg == "--" {
+                break;
+            }
+            match arg.as_str() {
+                "-c" => {
+                    let code = iter.next().ok_or(anyhow!("Missing code after -c"))?;
+                    codes.push(JsCode::Source(code));
+                }
+                "-b" => {
+                    let code = iter.next().ok_or(anyhow!("Missing code after -b"))?;
+                    let bytecode = hex::decode(code).context("Failed to decode bytecode")?;
+                    codes.push(JsCode::Bytecode(bytecode));
+                }
+                _ => {
+                    print_usage();
+                    bail!("Unknown option: {}", arg);
+                }
+            }
+        } else {
+            // File name
+            let code = std::fs::read_to_string(arg).context("Failed to read script file")?;
```

Comment:
What would happen if it's running in sidevm?

Reply:
Would panic with IoError.

Reply:
Got it